src/main/java/org/cactoos/io/LoggingOutputStream.java

```diff
@@ -0,0 +1,186 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Logged output stream.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ */
+@SuppressWarnings(
+    {
+        "PMD.AvoidDuplicateLiterals",
+        "PMD.LoggerIsNotStaticFinal"
+    }
+)
+public final class LoggingOutputStream extends OutputStream {
+
+    /**
+     * The output stream.
+     */
+    private final OutputStream origin;
+
+    /**
+     * Where the data comes goes.
+     */
+    private final String destination;
+
+    /**
+     * The logger.
+     */
+    private final Logger logger;
+
+    /**
+     * The bytes read.
+     */
+    private final AtomicLong bytes;
+
+    /**
+     * The time took to read.
+     */
+    private final AtomicLong time;
+
+    /**
+     * Ctor.
+     * @param output Destination of data
+     * @param dst The name of source data
+     */
+    public LoggingOutputStream(final OutputStream output, final String dst) {
+        this(output, dst, Logger.getLogger(dst));
+    }
+
+    /**
+     * Ctor.
+     * @param output Destination of data
+     * @param dst The name of source data
+     * @param lgr Message logger
+     */
+    public LoggingOutputStream(
+        final OutputStream output,
+        final String dst,
+        final Logger lgr
+    ) {
+        super();
+        this.origin = output;
+        this.destination = dst;
+        this.logger = lgr;
+        this.bytes = new AtomicLong();
+        this.time = new AtomicLong();
+    }
+
+    @Override
+    public void write(final int data) throws IOException {
+        final byte[] buf = {(byte) data};
+        this.write(buf, 0, 1);
+    }
+
+    @Override
+    public void write(final byte[] buf) throws IOException {
+        this.write(buf, 0, buf.length);
+    }
+
+    @Override
+    public void write(final byte[] buf, final int offset,
+        final int len) throws IOException {
+        final Instant start = Instant.now();
+        this.origin.write(buf, offset, len);
+        final Instant end = Instant.now();
+        final long millis = Duration.between(start, end).toMillis();
+        this.bytes.getAndAdd(len);
+        this.time.getAndAdd(millis);
+        final Level level = this.logger.getLevel();
+        if (!level.equals(Level.INFO)) {
+            this.logger.log(
```

Comment:
@fabriciofx same here, why pass its own level as an argument to itself? That level is encapsulated, there should be an overloaded method which doesnt ask for the level

Reply:
@amihaiemil The same explanation above.